dbms/src/Flash/Pipeline/Schedule/Tasks/Task.cpp

```diff
@@ -77,48 +78,57 @@ Task::Task(MemoryTrackerPtr mem_tracker_, const String & req_id)
     GET_METRIC(tiflash_pipeline_task_change_to_status, type_to_init).Increment();
 }
 
-ExecTaskStatus Task::execute() noexcept
+Task::~Task()
+{
+    RUNTIME_ASSERT(
+        exec_status == ExecTaskStatus::FINALIZE,
+        log,
+        "The state of the Task must be {} before it is destructed, but it is actually {}",
+        magic_enum::enum_name(ExecTaskStatus::FINALIZE),
+        magic_enum::enum_name(exec_status));
+}
+
+ExecTaskStatus Task::execute()
 {
     CHECK_FINISHED
     assert(getMemTracker().get() == current_memory_tracker);
```

Comment:
```suggestion
    RUNTIME_CHECK(getMemTracker().get() == current_memory_tracker);
```

Reply:
Here I still tend to use assert
1 is because the logic of set memory tracker is very simple
2 is worried that adding a RUNTIME_CHECK will affect the speed of await